tests/unit/test_k8s_service_info.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import pytest
+from charms.mlops_libs.v0.k8s_service_info import (
+    KubernetesServiceInfoObject,
+    KubernetesServiceInfoProvider,
+    KubernetesServiceInfoProviderWrapper,
+    KubernetesServiceInfoRelationDataMissingError,
+    KubernetesServiceInfoRelationMissingError,
+    KubernetesServiceInfoRequirer,
+    KubernetesServiceInfoRequirerWrapper,
+)
+from ops.charm import CharmBase
+from ops.model import TooManyRelatedAppsError
+from ops.testing import Harness
+
+TEST_RELATION_NAME = "test-relation"
+REQUIRER_CHARM_META = f"""
+name: requirer-test-charm
+requires:
+  {TEST_RELATION_NAME}:
+    interface: test-interface
+"""
+PROVIDER_CHARM_META = f"""
+name: provider-test-charm
+provides:
+  {TEST_RELATION_NAME}:
+    interface: test-interface
+"""
+
+
+class GenericCharm(CharmBase):
+    pass
+
+
+@pytest.fixture()
+def requirer_charm_harness():
+    return Harness(GenericCharm, meta=REQUIRER_CHARM_META)
+
+
+@pytest.fixture()
+def provider_charm_harness():
+    return Harness(GenericCharm, meta=PROVIDER_CHARM_META)
+
+
+def test_requirer_get_data_automatically_passes(requirer_charm_harness):
+    """Assert the relation data is as expected."""
+    # Initial configuration
+    requirer_charm_harness.set_model_name("test-model")
+    requirer_charm_harness.set_leader(True)
+    requirer_charm_harness.begin()
+
+    # Add and update relation
+    expected_data = KubernetesServiceInfoObject(name="some-service", port="7878")
+    data_dict = {"svc_name": expected_data.name, "svc_port": expected_data.port}
+    rel_id = requirer_charm_harness.add_relation(TEST_RELATION_NAME, "app", app_data=data_dict)
+
+    # Instantiate KubernetesServiceInfoRequirerWrapper class
+    requirer_charm_harness.charm._k8s_svc_info_requirer = KubernetesServiceInfoRequirer(
+        requirer_charm_harness.charm, relation_name=TEST_RELATION_NAME
+    )
+
+    # Get the relation data
+    actual_relation_data = requirer_charm_harness.get_relation_data(rel_id, "app")
+
+    # Assert returns dictionary with expected values
+    assert actual_relation_data["svc_name"] == expected_data.name
+    assert actual_relation_data["svc_port"] == expected_data.port
```

Comment:
I don't think this is testing anything about the lib.  This is asserting the data written directly to the relation during test setup is as expected

Reply:
I have updated the code and this test case, please let me know what you think.